export_snowflake/__init__.py

```
@@ -521,6 +528,134 @@ def flush_records(stream: str,
+                error_queue.put("ERROR")
+                return
+            
+        start_time = time.time()
+
+        s3_client.download_file(bucket, key, filename)
+                
+        # upload to snowflake
+        file_path = filename.replace("\\", "/")
+        db_sync = DbSync(config, schema, None, file_format_type)
+
+        s3_key = db_sync.put_to_stage(file_path, stream, temp_dir=None)
+            
+        # remove from local after uploading to snowflake
+        elasped_time = time.time() - start_time
+        LOGGER.info(f"Time taken to download and upload {filename} is {elasped_time}")
+        LOGGER.info(f'start_time: {start_time}, end_time: {time.time()} for {filename}')
+        os.remove(file_path)
+    except Exception as e:
+        LOGGER.info(
+            f"Error occurred in upload thread: {filename}")
+        # send exit signal to other thread
+        error_queue.put("ERROR")
+        raise e
+
+    
+
+def transfer_data_from_s3_to_snowflake(s3_csv_lists, config, schema, file_format_type):
+    parallel_transfer_start_time = time.time()
+    
+    error_queue = multiprocessing.Queue(1)
+    # boto3 clients are generally thread-safe, but not for resources or sessions
+    # https://boto3.amazonaws.com/v1/documentation/api/latest/guide/clients.html#multithreading-or-multiprocessing-with-clients
+    session = boto3.session.Session()
+    s3_client = session.client('s3')
+    
+    with concurrent.futures.ThreadPoolExecutor(max_workers=6) as executor:
+        futures = [executor.submit(upload_to_snowflake_per_thread, s3_client, csv_list, error_queue, config, schema, file_format_type) for csv_list in s3_csv_lists]
+        
+        for future in concurrent.futures.as_completed(futures):
+            try:
+                future.result()
+            except Exception as exc:
+                raise SymonException(f'An error occurred while loading data to Snowflake: ' + exc, 'snowflake.clientError')
```

Is this raising of error working? I noticed ThreadPool swallowing exceptions.